route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java

```diff
@@ -0,0 +1,248 @@
+package com.here.android.example.route.tta;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+import com.here.android.mpa.common.ApplicationContext;
+import com.here.android.mpa.common.MapEngine;
+import com.here.android.mpa.common.OnEngineInitListener;
+import com.here.android.mpa.guidance.TrafficUpdater;
+import com.here.android.mpa.mapping.Map;
+import com.here.android.mpa.mapping.MapRoute;
+import com.here.android.mpa.mapping.MapView;
+import com.here.android.mpa.routing.CoreRouter;
+import com.here.android.mpa.routing.DynamicPenalty;
+import com.here.android.mpa.routing.Route;
+import com.here.android.mpa.routing.RoutePlan;
+import com.here.android.mpa.routing.RouteResult;
+import com.here.android.mpa.routing.RouteTta;
+import com.here.android.mpa.routing.RoutingError;
+
+import java.io.File;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    private final int REQUEST_CODE = 0x754;
+
+    private MapView mapView;
```

Comment:
Try using code style that is used across all here android samples. E.g. class variables should have `m_` prefix, etc.